games-list/src/components/OneGame.tsx

```diff
@@ -1,51 +1,85 @@
+  gameObject: { game: string; complete: boolean; details: boolean };
   gamesList: Array<{ game: string; complete: boolean; details: boolean }>;
   setList: Dispatch<
     SetStateAction<{ game: string; complete: boolean; details: boolean }[]>
   >;
+  setModal: Dispatch<
+    SetStateAction<{
+      gameName: { game: string; complete: boolean; details: boolean };
+      open: boolean;
+    }>
+  >;
 }
+
 interface oneGameElement {
   game: string;
   complete: boolean;
   details: boolean;
 }
-
-export default function OneGame({ element, gamesList, setList }: OneGameProps) {
+/* This component signifies details related to any one game added and its components and modal */
+export default function OneGame({
+  gameObject,
+  gamesList,
+  setList,
+  setModal,
+}: OneGameProps) {
+  const [clicked, setClicked] = useState(false); //checks if game component is clicked
   let getName = (complete: boolean) =>
     complete ? styles.striked : styles.unstriked; // variable to changle css style used on text when clicked
-  let query = `search=${element}`;
   let gamePic =
     "https://media.rawg.io/media/games/1c3/1c305096502c475c00276c827f0fd697.jpg";
 
-  function handleClick(element: {
-    game: string;
-    complete: boolean;
-    details: boolean;
-  }) {
-    //removes element from list of games
-    setList(gamesList.filter((list) => list !== element));
+  function closeObject(gameObject: oneGameElement) {
+    //removes element from list of games when 'x' button is clicked
+    setModal({ gameName: gameObject, open: true });
   }
   function completeCheck(element: oneGameElement) {
-    //changes complete status of element object to true when x button is clicked
+    //changes complete status of element object to true when component is clicked
     setList(
       gamesList.map((game) =>
         game === element ? { ...game, complete: !element.complete } : game
       )
     );
+    setClicked(true);
   }
-  function revealDetails(element: oneGameElement) {
+  function revealDetails(gameObject: oneGameElement) {
     //changes details status of element to false when v button is clicked
     setList(
       gamesList.map((game) =>
-        game === element ? { ...game, details: !element.details } : game
+        game === gameObject ? { ...game, details: !gameObject.details } : game
       )
     );
   }
+
+  //Uploads data to mongo DB when game is added
+  async function uploadStriked() {
+    const response = await fetch("/api/replaceData", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ mongoList: gamesList }),
+    });
+    if (response.ok) {
+      return console.log(response);
+    } else {
+      alert("ERROR!!");
+    }
+  }
+  //used to prevent gamesList data being uploaded to DB before state is updated
+  useEffect(() => {
+    if (clicked) {
+      uploadStriked();
+      setClicked(false);
+    }
+  }, [clicked]);
```

Comment:
This entire logic of uploading data is...dubious. I would re-write it to be simpler, but for now you can leave it as is 😅

Reply:
haha I'll look more into how to write up API's in nextJS and saving data. Had difficulties figuring out how to capture data that has been set by a useState hook and capturing that updated list right after. 

Thats when I started using useEffect to make it work 😆